### DISCUSSION TITLE
Implement Facebook comment with Nextjs

### DISCUSSION THREAD

**Discussion by IRediTOTO** [vercel/next.js#14614] [Help]
Hi, I am trying to add facebook to my app. I code like below,  I am trying to lazy load FB comment by use dynamic, but I see Facebook comment re-render too often so I use useMemo. It run pretty ok, but if I Go to homepage and back to mypost page, FB comment disapear... Finding a solution to implement FB comment. Thank you. 
What I want to archive? 
1) Lazy load FB to get best speed 
2) Fb just render onetime for app performance
```
// mypost.js
const Comment = dynamic(
   () => import("../../../lib/facebook/comments-facebook"),
   { ssr: false } 
);
export default (props) => {
   // some codes...
   return (
      <TAB>
             <OtherTabs />
             <Comment />
      </TAB>
)}
```

Here is Facebook comment component comments-facebook.js

```
import { useEffect, memo } from "react";

const CommentsFacebook =(props) => {
   console.log("render CommentsFacebook")
   useEffect(() => {
      console.log("load facebook script")
      const facebookScript = document.createElement("script");
      facebookScript.async = true;
      facebookScript.src = `https://connect.facebook.net/en_US/sdk.js#xfbml=1&version=v7.0&appId=${process.env.FACEBOOK_APP_ID}&autoLogAppEvents=1`;
      document.body.appendChild(facebookScript);
   },[]);

   return (
      <React.Fragment>
         <div id="fb-root"></div>
         <div className="fb-comments" data-href={process.env.FACEBOOK_APP_ID} data-numposts="10" data-width="100%"> 
         </div>
      </React.Fragment>
   );
};

export default CommentsFacebook;
```

**Comment by kbdemba** [comment]
I had the same issue but this is how I solved mine

add the following code in useEffect in the comments component.
if(window.FB){
   window.FB.XFBML.parse();
}

Also, I think you should load the Facebook script once. So move it from comments-facebook.js file to  _app.js file.

**Reply by IRediTOTO** [reply]
Lets me try, look good

**Comment by dibasdauliya** [comment]
Hello @IRediTOTO  please let me know if you came up with any solution. I am having same problem.
https://github.com/vercel/next.js/discussions/30965

**Reply by IRediTOTO** [reply]
My solution is use facebook react package. Did u try?

**Reply by dibasdauliya** [reply]
> My solution is use facebook react package. Did u try?

I tried it but it didn't work. :(
https://codesandbox.io/s/crazy-gwen-hzvmx?file=/src/App.js

**Reply by IRediTOTO** [reply]
Can next script resve this problem?

**Reply by dibasdauliya** [reply]
> Can next script resve this problem?

Wrapping `<FacebookProvider.../>` with Next script? I guess no

**Reply by dibasdauliya** [reply]
> https://codesandbox.io/s/crazy-gwen-hzvmx?file=/src/App.js

Oh! I was using the wrong `href` value. It is working now. Thank you :)

**Reply by dibasdauliya** [reply]
Hello @IRediTOTO , I am using Facebook comments in this way 
![image](https://user-images.githubusercontent.com/51583004/140450739-50fb48c7-1900-490e-a647-0c177a2fa93b.png)

and I did one comment and it is showing the same comment everywhere until I refresh.

**Reply by jsonpreet** [reply]
Hi,

Your problem is solved?
